src/test/movinginterquartilemean_test.cpp

```diff
@@ -109,4 +112,28 @@ TEST_F(MovingInterquartileMeanTest, doubles9) {
     }
 }
 
+void BM_MovingIQM_Insertion(benchmark::State& state) {
+    std::mt19937 gen; // explicitly don't seed for reproducibility
```

Comment:
Why make it not reproducible? This makes future debbuging really hard.
In my opinion a test case shall always use the same data. If you want increase the coverage, you should increase the test data size instead of making it random.

Reply:
I 100% agree. Its explicitly **not** initialized with a seed (usually you would leverage `std::random_device` for that). So its seed is implicitly 0 (which I could make explicit I guess), so it will always generate the same sequence of numbers and is thus reproducible.
Thinking about this `uniform_real_distribution` is not really realistic, instead it `normal_distribution` would likely make more sense at least when thinking about the TapFilter usecase specifically.

Reply:
Ah, than I just misunderstood the wording

Reply:
See the [doc for the mersenne_twister_engine](https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine/mersenne_twister_engine):

> The default constructor.
> * If the default-constructed engine is of type std::mt19937, the 10000th consecutive invocation of it produces the value 4123659995.